packages/keycloak-user-management/src/components/forms/UserForm/utils.tsx

```diff
@@ -228,3 +228,23 @@ export const getUserFormPayload = (values: FormFields) => {
     userGroups: values.userGroups ?? [],
   };
 };
+
+/**
+ * converts userGroups to options that can be passed to antd Select
+ *
+ * @param userGroups - the userGroups in an array
+ */
+export const getUserGroupsOptions = (userGroups: UserGroup[]): SelectOption[] =>
+  userGroups.map((userGroup) => {
+    return { value: userGroup.id, label: userGroup.name };
+  });
+
+/**
+ * filters options listed once user types in userGroups select
+ *
+ * @param inputValue - the search text,
+ * @param option - one of the options
+ */
+export const userGroupOptionsFilter = (inputValue: string, option?: SelectOption) => {
+  return !!option?.label.toLowerCase().includes(inputValue.toLowerCase());
+};
```

Comment:
option.label is sometimes undefined which would throw an error here. See https://github.com/opensrp/web/issues/692

Reply:
@machariamuguku does this mean we can have userGroups without names, because that's the only situation where the label would be undefined

Reply:
@p-netm idk why it happens but instead of option.label it sometimes returns option.children. That's why I had a hard time typing this [function](https://github.com/opensrp/web/blob/eb48d21d8f6b4cc4e4adc291a0480db65ff1489b/packages/team-assignment/src/components/ListView/index.tsx#L185)

Reply:
@machariamuguku I modified how the options are passed to the [Select](https://github.com/opensrp/web/blob/e475fd6b6036fe07ba3eeb1847a1d17698b43995/packages/keycloak-user-management/src/components/forms/UserForm/index.tsx#L195) such that they are of type `SelectOption`, antd Select passes each of these option as `SelectOption` to the second argument of the filterOptions prop function.

 I think option.children during filtering is needed if you actually create the Select.Options jsx manually.

not sure if this makes sense